app/models/subnet.rb

```diff
@@ -38,6 +38,7 @@ class Subnet < ActiveRecord::Base
   validates :boot_mode, :inclusion => BOOT_MODES.values
   validates :ipam, :inclusion => IPAM_MODES.values
   validates :name,    :length => {:maximum => 255}, :uniqueness => true
+  validates :dns, :tftp, :dhcp, :proxy_features => { :dns => "DNS", :tftp => "TFTP", :dhcp => "DHCP" }
```

Comment:
This syntax is strange for a validator, as it's specifying different options for different fields.  The options should apply to all fields, e.g.

    validates :dns, :proxy_feature => 'DNS'

and then hypothetically, if you had the same feature required for two fields, it'd be:

    validates :dns1, :dns2, :proxy_feature => 'DNS'